ExtractUserDataActivity/handler.ts

```diff
@@ -292,32 +294,27 @@ export const findNotificationsForAllMessages = (
   ActivityResultQueryFailure,
   ReadonlyArray<RetrievedNotification>
 > =>
-  array.sequence(taskEither)(
-    messages.map(m =>
-      notificationModel.findNotificationForMessage(m.id).foldTaskEither(
-        e =>
-          fromLeft(
-            ActivityResultQueryFailure.encode({
-              kind: "QUERY_FAILURE",
-              reason: `notificationModel.findNotificationForMessage| ${
-                e.kind
-              }, ${getMessageFromCosmosErrors(e)}`
-            })
-          ),
-        maybeNotification =>
-          maybeNotification.foldL(
-            () =>
-              fromLeft(
-                ActivityResultQueryFailure.encode({
-                  kind: "QUERY_FAILURE",
-                  reason: `notificationModel.findNotificationForMessage| No notifications found for messageId ${m.id}`
-                })
-              ),
-            notification => fromEither(right(notification))
-          )
+  array
+    .sequence(taskEitherSeq)(
+      messages.map(m =>
+        notificationModel.findNotificationForMessage(m.id).mapLeft(e =>
+          ActivityResultQueryFailure.encode({
+            kind: "QUERY_FAILURE",
+            reason: `notificationModel.findNotificationForMessage| ${
+              e.kind
+            }, ${getMessageFromCosmosErrors(e)}`
+          })
+        )
       )
     )
-  );
+    .map(arrayOfMabeNotifications =>
+      arrayOfMabeNotifications
```

Comment:
you can use `array.catOptions` === filter(isSome) + x.value

https://github.com/gcanti/fp-ts/blob/1.17.4/src/Array.ts#L1312